src/main/java/com/google/sps/dao/AssignmentDao.java

```diff
@@ -89,4 +94,112 @@ public ValidationResponse createAssignment(Assignment assignment) {
         }
         return assignments;
     }
+
+    @Override
+    @SneakyThrows
+    public ValidationResponse updateAssignmentForStudent(Assignment newAssignment) {
+
+        ValidationResponse.Builder validationResponseBuilder = null;
+
+        Assignment oldAssignment = getAssignmentById(newAssignment.getId());
+        newAssignment.setId(oldAssignment.getId());
+        newAssignment.setUpdated(System.currentTimeMillis());
+
+        if (ValidationUtils.isEmptyOrWhiteSpace(newAssignment.getSolution())) {
+            validationResponseBuilder = ValidationResponse.builder();
+            validationResponseBuilder.status(ValidationErrors.STATUS_NOT_OK);
+            validationResponseBuilder.message(ValidationErrors.MESSAGE_NOT_OK);
+
+            validationResponseBuilder.error(
+                    ValidationErrors.isEmptyOrWhiteSpace(Assignment.Keys.SOLUTION)
+            );
+        }
+
+        //do not set scored_marks attribute
+        //solution attribute is set from the request
+        newAssignment.setCreated(oldAssignment.getCreated());
```

Comment:
Since old assignment contains everything except the solution it will be easier to update the solution, submitted and updated fields of oldAssignment and use oldAssignment to update the db.

Reply:
done